plugins/chromatic_correction.py

```diff
@@ -0,0 +1,252 @@
+            matched_1.add(i)
+            matched_2.add(j)
+
+    return numpy.array(pairs)
+
+
+class ChromaticCorrectionPlugin(Plugin):
+    name = "Chromatic Correction"
+    __version__ = "1.0"
+    __author__ = "Karishma Kumar"
+    __license__ = "GPLv2"
+
+    def __init__(self, microscope, main_app):
+        if "cryosecom-localization" in main_app.main_data.tab.choices.values():
+            super().__init__(microscope, main_app)
+            self.addMenu("Data correction/Lateral chromatic correction", self.start)
+
+    def show_static_stream(self, filename: str, data: model.DataArray):
+        """
+        Save the given data, read and display in the analysis tab
+        :param filename: filename of the data array
+        :param data: value and metadata of the given data array
+        """
+        # save the image
+        dataio.tiff.export(filename, data)
+        # Read the saved images and display
+        analysis_tab = self.main_app.main_data.getTabByName('analysis')
+        self.main_app.main_data.tab.value = analysis_tab
+        analysis_tab.load_data(filename, extend=True)
+
+    def compute_transformation(self, ref_channel, channels, max_distance):
+        """
+        Compute affine transformation between a reference channel and another coloured channel.
+        :param ref_channel: (model.DataArray) The reference channels to which other channels are aligned
+        :param channels: (list of model.DataArray) The channels are transformed to align with reference channel
+        :param max_distance: (float in meters) The maximum distance accepted to match two same locations in between
+         reference channel and channels
+        """
+        trans_per_channel = {model.MD_CHROMATIC_COR: {}}  # stores transformation values per channel
+        dist_pxl = int(max_distance / ref_channel.metadata[model.MD_PIXEL_SIZE][0])  # convert m to pixel units
+        centers_ref = MaximaFind(ref_channel, BEAD_QUANTITY, dist_pxl)
+
+        for i, im in enumerate(channels):
+            # Find corresponding points
+            centers_new = MaximaFind(im, BEAD_QUANTITY, dist_pxl)
+            corresponding_pairs = find_corresponding_points(centers_ref, centers_new, dist_pxl)
+
+            # Extract the matching points
+            points1 = centers_ref[corresponding_pairs[:, 0]]
+            points2 = centers_new[corresponding_pairs[:, 1]]
+
+            if len(corresponding_pairs) < 4:
+                txt = ('The chromatic correction did not happen for channel with %s. Skipping this channel.\n\n'
+                       'Alternatively, navigate to location with more beads or change the bead diameter in the script'
+                       % im.metadata[model.MD_DESCRIPTION])
+                box = wx.MessageDialog(self.main_app.main_frame,
+                                       txt, "Partial failure of chromatic correction", wx.OK)
+                box.ShowModal()
+                box.Destroy()
+
+            else:
```

Comment:
Instead of a `else`, place a `continue` at the end of the `if` block, and here, the "normal" code can be one indentation less, as if there was no error handling.